components/common-go/watch/file.go

```diff
@@ -83,23 +90,40 @@ func File(ctx context.Context, path string, onChange func()) error {
 					continue
 				}
 
-				currentHash, err := hashConfig(path)
-				if err != nil {
-					log.WithError(err).Warn("Cannot check if config has changed")
-					return
+				currentHash, hashErr := hashConfig(path)
+				if hashErr != nil {
+					log.WithError(hashErr).WithField("event", event.Name).Warn("Cannot check if config has changed, backing off")
+
+					if currentBackoff == 0 {
+						currentBackoff = initialBackoff
+					} else {
+						currentBackoff *= 2
+						if currentBackoff > maxBackoff {
+							currentBackoff = maxBackoff
+						}
+					}
+
+					select {
+					case <-time.After(currentBackoff):
```

Comment:
Curious: Why do we need the backoff in the first place? Why can't we just not return an error? :thinking:

Reply:
Oh, I see - we do now, so that's all set :) 

I just made a couple changes:
* [176c98b](https://github.com/gitpod-io/gitpod/pull/20752/commits/176c98ba4af83f7a676724f0db4114712d8cad2f)
* [7a14aef](https://github.com/gitpod-io/gitpod/pull/20752/commits/7a14aef6a93bec45f77c1733284dd8a5ef15596d)

And will test again in a bit.